src/sharedOptions.ts

```diff
@@ -18,6 +18,11 @@ export interface LifeCycleEventsMap<ResponseType> {
   'request:start': (request: MockedRequest) => void
   'request:match': (request: MockedRequest) => void
   'request:unhandled': (request: MockedRequest) => void
+  'request:unhandledException': (
+    request: MockedRequest,
```

Comment:
What do you think if me model the order of arguments based on the developer's expectations of them?

I think when handling exceptions, developers are most interested in the error thrown, so I'd recommend it go first. 

```ts
'request:unhandledException': (error: Error, request: MockedRequest, handler: RequestHandler) => void
```